jupyterlab/labhubapp.py

```diff
@@ -18,7 +19,29 @@
     from jupyterhub.singleuser.mixins import make_singleuser_app
 except ImportError:
     # backward-compat with jupyterhub < 1.3
-    from jupyterhub.singleuser import SingleUserNotebookApp as SingleUserServerApp
+    try:
+        from jupyterhub.singleuser import SingleUserNotebookApp as SingleUserServerApp
+    except ImportError as e:
+        # jupyterhub is not installed at all
+        venv_info = sys.prefix
+        is_venv = hasattr(sys, "real_prefix") or (
+            hasattr(sys, "base_prefix") and sys.base_prefix != sys.prefix
+        )
```

Comment:
FWIW the conditional can be simplified to:

```suggestion
        is_venv = sys.base_prefix != sys.prefix
```

As this project only supports Python 3.9+ anyway. The rest of the check is only necessary for very old Python/virtualenv versions.